sql/core/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -819,7 +819,13 @@ private[sql] class SQLConf extends Serializable with CatalystConf with Logging {
 
   def variableSubstituteDepth: Int = getConf(VARIABLE_SUBSTITUTE_DEPTH)
 
-  def warehousePath: String = new Path(getConf(StaticSQLConf.WAREHOUSE_PATH)).toString
+  def warehousePath: String = {
+    if (contains(StaticSQLConf.WAREHOUSE_PATH.key)) {
```

Comment:
What is the reason we are not doing the same check, as shown in [another place](https://github.com/shivaram/spark-1/blob/6eec97d463b027c93b48621b55e3bd4005dc0f7e/sql/core/src/main/scala/org/apache/spark/sql/internal/SharedState.scala#L59)?